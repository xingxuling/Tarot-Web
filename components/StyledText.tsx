import React from 'react';
import { Text, TextProps, StyleSheet } from 'react-native';

export function MonoText(props: TextProps) {
  return <Text {...props} style={[props.style, styles.mono]} />;
}

const styles = StyleSheet.create({
  mono: {
    fontFamily: 'SpaceMono'
  }
});
