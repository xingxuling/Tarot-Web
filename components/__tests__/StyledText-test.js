/* eslint-env jest */
import * as React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';
import { MonoText } from '../StyledText';

describe('MonoText', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <MonoText>
        <Text>Snapshot test!</Text>
      </MonoText>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
