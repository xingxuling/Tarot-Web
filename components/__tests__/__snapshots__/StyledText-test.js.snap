// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonoText renders correctly 1`] = `
<Text
  style={
    [
      undefined,
      {
        "fontFamily": "SpaceMono",
      },
    ]
  }
>
  <Text>
    Snapshot test!
  </Text>
</Text>
`;
